cabal-install-solver/src/Distribution/Solver/Modular/MessageUtils.hs

```diff
@@ -0,0 +1,57 @@
+-- | Utility functions providing extra context to cabal error messages
+
+module Distribution.Solver.Modular.MessageUtils (
+  allKnownExtensions,
+  cutoffRange,
+  mostSimilarElement,
+  showUnsupportedExtension,
+  showUnsupportedLanguage,
+  withinRange
+) where
+
+import Data.Foldable (minimumBy)
+import Data.Ord (comparing)
+import Distribution.Pretty (prettyShow) -- from Cabal
+import Language.Haskell.Extension
+         ( Extension(..), Language(..), knownLanguages, knownExtensions )
+import Text.EditDistance ( defaultEditCosts, levenshteinDistance )
+import Text.Printf (printf)
+
+showUnsupportedExtension :: Extension -> String
+showUnsupportedExtension (EnableExtension knownExt) = prettyShow knownExt ++ "which is not supported"
+showUnsupportedExtension (DisableExtension knownExt) = prettyShow knownExt ++ "which is not supported"
+showUnsupportedExtension (UnknownExtension extStr) = formatMessage cutoffRange "extension" extStr (mostSimilarElement extStr allKnownExtensions)
+
+showUnsupportedLanguage :: Language -> String
+showUnsupportedLanguage (UnknownLanguage langStr) = formatMessage cutoffRange "language" langStr (mostSimilarElement langStr (show <$> knownLanguages))
+showUnsupportedLanguage knownLanguage = prettyShow knownLanguage ++ "which is not supported"
+
+allKnownExtensions :: [String]
+allKnownExtensions = enabledExtensions ++ disabledExtensions
+  where
+    enabledExtensions = show <$> knownExtensions
+    disabledExtensions = ("No" ++) . show <$> knownExtensions
```

Comment:
use `prettyShow` on `Extension`, i.e. `map (prettyShow . EnableExtension) knownExtensions` and `DisableExtension`. Using `"No"++` is wrong.

Reply:
Fixed